src/input/src/interactions/toml.rs

```diff
@@ -63,6 +63,25 @@ impl FromToml for LennardJones {
     }
 }
 
+impl FromToml for Mie {
+    fn from_toml(table: &Table) -> Result<Mie> {
+        let sigma = try_extract_parameter!(table, "sigma", "Mie potential");
+        let epsilon = try_extract_parameter!(table, "epsilon", "Mie potential");
+        let m = try_extract_parameter!(table, "m", "Mie potential");
+        let n = try_extract_parameter!(table, "n", "Mie potential");
```

Comment:
You can use `extract::str` and `extract::number` here instead

Reply:
So I realised that the rest of the code did not used these functions, and made #236. But this PR and #236 will conflict, so we need to merge one and then rebase the other. I am fine with merging this PR with `try_extract_parameter!`, and then rebasing #236 :smiley: